index.js

```diff
@@ -75,46 +44,45 @@ idle.addListener = function (intervalSec, callback) {
         lastCheckDateMsec = Date.now();
 
         // ask the system what's the idle time
-        idle.tick(function(idleSeconds, error){
-
-            if(error) {
-                callback({ id: listenerIndex }, error);
-                timeoutId = setTimeout(checkIsAway, defaultIntervalMsec);
-                return;
-            }
-
-            // is aways if the idle duration is bigger than a interval duration fraction
-            isAway = (idleSeconds * 1000) >= (intervalDurationMsec * INTERVAL_AWAY_IDLE_TIME_PERCENTAGE);
-
-            if(!isAfk && isAway) {
-                callback({
-                	status: 'away',
-                	seconds: idleSeconds,
-                	id: listenerIndex
-                });
-
-                isAfk = true;
-            }
-            else if(isAfk && !isAway) {
-                callback({
-                    status: 'back',
-                    seconds: idleSeconds,
-                    id: listenerIndex
-                });
-
-                isAfk = false;
-            }
-
-            timeoutId = setTimeout(checkIsAway, isAfk ? AFK_SYSTEM_POLLING_TIMEOUT_MSEC : defaultIntervalMsec);
-        });
+        try {
+            idleSeconds = tick();
+        } catch (err) {
+            if (typeof errorHandler !== 'undefined') errorHandler(err);
```

Comment:
this is a breaking change:
the error before was propagated as the second argument `callback({ id: listenerIndex }, error);`
Do you want to keep it like this?